src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureCount = 5;
+        int rnd = random.nextInt(figureCount);
+        int valueCount = 100;
+        switch (rnd) {
+            case 0:
+                return new Circle(random.nextInt(valueCount), color.getRandomColor());
+            case 1:
+                return new IsoscelesTrapezoid(random.nextInt(valueCount),
+                        random.nextInt(valueCount),
+                        random.nextInt(valueCount),
+                        color.getRandomColor());
+            case 2:
+                return new Rectangle(random.nextInt(valueCount),
+                        random.nextInt(valueCount),
+                        color.getRandomColor());
+            case 3:
+                return new RightTriangle(random.nextInt(valueCount),
+                        random.nextInt(valueCount),
+                        color.getRandomColor());
+            case 4:
+                return new Square(random.nextInt(valueCount), color.getRandomColor());
+            default:
+                getDefaultFigure();
+        }
+        return getDefaultFigure();
+    }
+
+    public Figure getDefaultFigure() {
+        final int defaultValue = 10;
```

Comment:
same for 10 and WHITE here

Reply:
not fixed
this should be class-level constant (add static + naming)
https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#all-magic-numbers-in-your-code-should-be-constants